google-cloud-storage/acceptance/storage/bucket_uniform_bucket_level_access_test.rb

```diff
@@ -0,0 +1,205 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "storage_helper"
+
+describe Google::Cloud::Storage::Bucket, :uniform_bucket_level_access, :storage do
+  let(:bucket_name) { $bucket_names.first }
+  let :bucket do
+    storage.bucket(bucket_name) ||
+    safe_gcs_execute { storage.create_bucket(bucket_name) }
+  end
+
+  let(:files) do
+    { logo: { path: "acceptance/data/CloudPlatform_128px_Retina.png" },
+      big:  { path: "acceptance/data/three-mb-file.tif" } }
+  end
+  let(:local_file) { File.new files[:logo][:path] }
+
+  let(:user_val) { "user-blowmage@gmail.com" }
+
+  after do
+    # always reset the uniform_bucket_level_access
+    # always reset the bucket permissions
+    bucket.uniform_bucket_level_access = false if bucket.uniform_bucket_level_access?
+    bucket.default_acl.private!
+    bucket.files.all { |f| f.delete rescue nil }
+  end
+
+  it "sets uniform_bucket_level_access true and is unable to modify file ACL rules" do
+    refute bucket.uniform_bucket_level_access?
+    bucket.uniform_bucket_level_access_locked_at.must_be :nil?
+    file = bucket.create_file local_file, "ReaderTest.png"
+
+    bucket.uniform_bucket_level_access = true
+    assert bucket.uniform_bucket_level_access?
+    bucket.uniform_bucket_level_access_locked_at.must_be_kind_of DateTime
+
+    err = expect do
+      file.acl.add_reader user_val
+    end.must_raise Google::Cloud::PermissionDeniedError
+    err.message.must_match /does not have storage.objects.get access to/
+  end
+
+  it "sets uniform_bucket_level_access true and is unable to get the file" do
+    refute bucket.uniform_bucket_level_access?
+    file = bucket.create_file local_file, "ReaderTest.png"
+
+    bucket.uniform_bucket_level_access = true
+    assert bucket.uniform_bucket_level_access?
+    err = expect do
+      file.reload!
+    end.must_raise Google::Cloud::PermissionDeniedError
+    err.message.must_match /does not have storage.objects.get access to/
+
+    bucket.uniform_bucket_level_access = false
+    file.reload!
+  end
+
+  it "sets uniform_bucket_level_access true and is unable to modify bucket ACL rules" do
+    refute bucket.uniform_bucket_level_access?
+    bucket.uniform_bucket_level_access = true
+    assert bucket.uniform_bucket_level_access?
+    err = expect do
+      bucket.acl.public!
+    end.must_raise Google::Cloud::InvalidArgumentError
+    err.message.must_match /Cannot use ACL API to update bucket policy when Bucket Policy Only is enabled./
```

Comment:
Please remove this and other checks against the exact error message. The message will change in the future so we don't want the tests to fail when it does. It seems sufficient to me to make sure that calls to update ACL when UBLA is on fail in general